snapcraft/extractors/_metadata.py

```diff
@@ -0,0 +1,78 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import yaml
+from typing import Any, Dict
+
+
+class ExtractedMetadata(yaml.YAMLObject):
+    """Collection of metadata extracted from a part."""
+
+    yaml_tag = u'!ExtractedMetadata'
+
+    def __init__(self, *, summary='', description='') -> None:
+        """Create a new ExtractedMetadata instance.
+
+        :param str summary: Extracted summary
+        :param str description: Extracted description
+        """
+
+        self._data = {}  # type: Dict[str, str]
+
+        if summary:
+            self._data['summary'] = summary
+        if description:
+            self._data['description'] = description
```

Comment:
what is the story with blobs like icon and desktop file?

Reply:
As we briefly discussed, we can base64 them, or save file paths here and grab them when creating the meta/ dir.